client/allocrunner/taskrunner/template/template.go

```diff
@@ -635,13 +643,68 @@ func newRunnerConfig(config *TaskTemplateManagerConfig,
 	}
 	conf.Templates = &flat
 
+	// TODO: Code review this. It seems that this setting is only used to force
+	// errors in a specific test scenarios. We need to rethink implementation.
```

Comment:
For this review. I don't know if this is safe as is, or if my changes should result in us getting rid of this test-only setting, and just using the retry I've now defined.

Reply:
I refactored out the special testing only field and used the newly added RetryConfig since they were both used to manage the same Consul Template settings.